src/common/k8s-api/kube-api.ts

```diff
@@ -310,15 +353,20 @@ export class KubeApi<
 
     for (const apiUrl of apiBases) {
       try {
-        // Split e.g. "/apis/extensions/v1beta1/ingresses" to parts
-        const { apiPrefix, apiGroup, apiVersionWithGroup, resource } = parseKubeApi(apiUrl);
-
-        // Request available resources
-        const { resources } = (await this.request.get(`${apiPrefix}/${apiVersionWithGroup}`)) as unknown as KubeApiResourceList;
-
-        // If the resource is found in the group, use this apiUrl
-        if (resources.find(({ name }) => name === resource)) {
-          return { apiPrefix, apiGroup };
+        const { apiPrefix, apiGroup, resource } = parseKubeApi(apiUrl);
+        const list = await this.request.get(`${apiPrefix}/${apiGroup}`) as KubeApiResourceVersionList;
+        const resourceVersions = getOrderedVersions(list);
+
+        for (const resourceVersion of resourceVersions) {
+          const { resources } = await this.request.get(`${apiPrefix}/${resourceVersion.groupVersion}`) as KubeApiResourceList;
+
+          if (resources.some(({ name }) => name === resource)) {
+            return {
+              apiPrefix,
+              apiGroup,
+              apiVersionPreferred: resourceVersion.version,
+            };
+          }
         }
```

Comment:
Could we use `Promise.race` here, or is the intended behaviour to sequentially probe for requests until bingo?

Reply:
The intended behaviour is to sequentially probe. Furthermore this only happens the very first time any call occurs.